src/components/form/submission.vue

```diff
@@ -79,7 +85,16 @@ const fetchForm = () => {
   }).catch(noop);
 };
 
-fetchForm();
+// To facilate validateData in route guard
+if (props.projectId) {
+  fetchProject().then(() => {
+    fetchForm();
```

Comment:
Is there a way to send these two requests at once, but hide the component until the project response is received? I feel like we're setting up a series of dependent requests, which could slow down page load:

- The project is requested
- The form is requested only once the project response is received
- The form XML is requested only once the form response is received

I'm thinking of something like:

```vue
<template>
  <loading :state="initiallyLoading"/>
  <component :is="component" v-if="dataExists && component != null" v-bind="bindings"/>
<template>

<script setup>
// ...

const { initiallyLoading, dataExists } = resourceStates(props.projectId ? [project, form] : [form]);

if (props.projectId) fetchProject();
fetchForm();
</script>
```